src/main/proto/netflix/titus/titus_master_api.proto

```diff
@@ -0,0 +1,113 @@
+syntax = "proto3";
+
+// TitusMaster Supervisor API specification. This API provides access to TitusMaster operational state
+// (leader/non-leader, activation time, etc), as well as provides management operations.
+//
+
+package com.netflix.titus;
+
+import "google/protobuf/empty.proto";
+
+option java_multiple_files = true;
+option java_package = "com.netflix.titus.grpc.protogen";
+option java_outer_classname = "TitusMasterProto";
+
+option go_package = "titus";
+
+// ----------------------------------------------------------------------------
+// Titus supervisor data model.
+
+message MasterStatus {
+
+    enum MasterState {
+
+        /// TitusMaster instance is starting, and its operational state is not known yet.
+        Starting = 0;
+
+        /// TitusMaster is inactive, and does not participate in the leader election process. Predefined
+        //  reason codes in this state:
+        //  * 'outOfService' - the instance is healthy, but taken of out of service by an administrator
+        //  * 'unhealthy' - the instance is not healthy
+        Inactive = 1;
+
+        /// TitusMaster is healthy, and participates in the leader election process, but is not the leader yet.
+        NonLeader = 2;
+
+        /// TitusMaster bocame a leader, and is activating all components. It does not accept client requests yet.
+        LeaderActivating = 3;
+
+        /// TitusMaster is a fully activated leader.
+        LeaderActivated = 4;
+    }
+
+    /// (Required) TitusMaster instance state
+    MasterState state = 1;
+
+    /// (Optional) Detailed code for being in a specific state.
+    string reasonCode = 2;
+
+    /// (Optional) Textual description accompanying the 'reasonCode'.
+    string reasonMessage = 3;
+
+    /// Time when a transition to a state happened.
+    uint64 timestamp = 4;
+
+}
+
+/// A descriptor of a single TitusMaster instance.
+message MasterInstance {
+
+    string instanceId = 1;
+
+    string ipAddress = 2;
+
+    MasterStatus status = 3;
+
+    repeated MasterStatus statusHistory = 4;
+}
+
+message SupervisorEvent {
+    message MasterInstanceUpdate {
+        MasterInstance instance = 1;
+    }
+
+    message MasterInstanceRemoved {
+        string instanceId = 1;
+    }
+
+    oneof Event {
+        MasterInstanceUpdate masterInstanceUpdate = 1;
+        MasterInstanceRemoved masterInstanceRemoved = 2;
+    }
+}
+
+// ----------------------------------------------------------------------------
+// GRPC services
+
+message MasterInstanceId {
+    string instanceId = 1;
+}
+
+message MasterInstances {
+    repeated MasterInstance instances = 1;
+}
+
+service SupervisorService {
```

Comment:
It looks like some methods on this service are intended to be called on a single instance, and others are routed to the correct instance, regardless of which instance gets the request.

For consistency, should we make them all route to the correct instance? i.e.: change `StopBeingLeader()` to also receive a `MasterInstanceId` as a parameter?

Reply:
There is no routing. Each Master has full information about all the other instances.

Reply:
I could remove the `GetMasterInstance(<id>)` method. As there are few master instances, fetching full list should be good enough for all cases.

Reply:
I think I understand this better after reviewing Netflix/titus-control-plane#340. Maybe just add a few comments explaining how we rely on gateway to find the current leader and call these methods directly on it?